horovod/common/operations.cc

```diff
@@ -865,6 +898,18 @@ void PerformOperation(TensorTable& tensor_table, MPIResponse response) {
 
         ACTIVITY_END_ALL(entries, timeline)
       }
+#elif HOROVOD_GPU_ALLREDUCE == 'D'
+      if (!horovod_global.ddl_initialized) {
+        // Initialize DDL
+        auto ddl_options = std::getenv("DDL_OPTIONS");
+        if (ddl_options == nullptr) {
+          throw std::logic_error("DDL_OPTIONS env variable needs "
+                                 "to be set to use DDL.");
+        }
+        DDL_CHECK(entries, "ddl_init", ddl_init(ddl_options))
+        horovod_global.ddl_initialized = true;
```

Comment:
One more question.  Does this pass `test_horovod_allreduce_multi_gpu()`?  In the case of NCCL, we dynamically create a communicator based on the set of GPUs that will participate in the operation.  How does DDL handle that?

Reply:
It probably won't pass this test. Right now, DDL handles only one device per process.

Reply:
@alsrgv I am adding a check for the "test_horovod_allreduce_multi_gpu" to fail gracefully. Just curious, what is the use case for Horovod to support several devices per process?

Reply:
I pushed the check.